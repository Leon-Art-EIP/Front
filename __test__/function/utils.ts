import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
