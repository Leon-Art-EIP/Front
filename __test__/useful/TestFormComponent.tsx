import { FormProvider } from "react-hook-form";
import type { ReactNode } from "react";
import useTestForm from "./useTestForm";

interface ITestInputProps {
  name: string;
  value?: string[] | string;
  children: ReactNode;
}

export default function TestFormComponent(props: ITestInputProps): JSX.Element {
  const methods = useTestForm({ [props.name]: props.value });

  return <FormProvider {...methods}>{props.children}</FormProvider>;
}
