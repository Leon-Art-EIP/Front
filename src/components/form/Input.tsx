import { HTMLInputTypeAttribute, useState } from "react";
import { useController } from "react-hook-form";
import { cn } from "../../tools/cn";
import VisibilityIcon from "@mui/icons-material/Visibility";
import VisibilityOffIcon from "@mui/icons-material/VisibilityOff";

interface IInputProps {
  type: HTMLInputTypeAttribute;
  name: string;
  className?: string;
  errorClassName?: string;
  placeholder: string;
  title?: string;
  hideError?: boolean;
  onBlur?: () => void;
  onFocus?: () => void;
}

export default function Input(props: IInputProps): JSX.Element {
  const {
    field: { onChange, value },
    fieldState: { error },
  } = useController({ name: props.name });

  const [showPassword, setShowPassword] = useState(false);
  const isPasswordType = props.type === "password";
  const inputType = isPasswordType && showPassword ? "text" : props.type;

  return (
    <div className="flex flex-col gap-2 sm:min-w-min min-w-0 relative">
      {props.title && <div className="text-lg text-tertiary">{props.title}</div>}
      <div className="relative">
        <input
          type={inputType}
          name={props.name}
          id={props.name}
          className={cn(
            "sm:min-w-min min-w-0 px-6 py-4 w-full rounded placeholder:text-tertiary-hover pr-10",
            props.className
          )}
          placeholder={props.placeholder}
          onChange={onChange}
          onBlur={props.onBlur}
          onFocus={props.onFocus}
          value={value}
          autoComplete="off"
        />
        {isPasswordType && (
          <button
            type="button"
            className="absolute inset-y-0 right-3 flex items-center"
            onClick={() => setShowPassword(!showPassword)}
            aria-label={showPassword ? "Masquer le mot de passe" : "Voir le mot de passe"}
          >
            {showPassword ? <VisibilityIcon /> : <VisibilityOffIcon />}
          </button>
        )}
      </div>
      {error && !props.hideError && <div className={cn("text-primary", props.errorClassName)}>{error.message}</div>}
    </div>
  );
}
