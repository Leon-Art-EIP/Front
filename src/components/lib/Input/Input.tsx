import React from "react";

/* c8 ignore start */
export interface InputProps {
  value?: string | number;
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
  type?: string;
  required?: boolean;
  placeHolder?: string;
  className?: string;
  label?: string;
  disabled?: boolean;
}

const Input: React.FC<InputProps> = ({
  value,
  onChange,
  type = "text",
  required = false,
  placeHolder,
  className = "",
  label = "",
  disabled = false,
}) => {
  return (
    <label className={`${className} flex flex-col gap-2 text-black font-regular text-md`}>
      <span>
        {label} {required && <span className="text-red-500">*</span>}
      </span>
      <input
        type={type}
        className="text-md rounded-xl p-3 bg-gray-700 border-gray-600 placeholder-[#8F8F8F] text-white disabled:cursor-not-allowed disabled:text-gray"
        placeholder={placeHolder}
        value={value}
        onChange={onChange}
        required={required}
        disabled={disabled}
      />
    </label>
  );
};

export default Input;
/* c8 ignore stop */
