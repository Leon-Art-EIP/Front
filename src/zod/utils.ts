export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
