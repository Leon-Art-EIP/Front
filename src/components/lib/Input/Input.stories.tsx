import type { Meta, StoryObj } from '@storybook/react';

import Input, { InputProps } from './Input';
import { ChangeEvent } from "react";

export default {
  title: 'Components/Input',
  component: Input,
} as Meta<typeof Input>;

type Story = StoryObj<InputProps>;

export const InputMock: Story = {
  args: {
    value: '',
    placeHolder: 'Enter some text here',
    onChange: (event: ChangeEvent<HTMLInputElement>) => console.log(event.target.value),
    label: 'Input',
    disabled: false,
    required: false,
    type: 'text',
  },
};
