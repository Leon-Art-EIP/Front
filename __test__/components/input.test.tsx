import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import TestFormComponent from "../useful/TestFormComponent";
import Input from "../../src/components/form/Input";

test("Input snapshot", () => {
  testSnapshot(
    <TestFormComponent name="input">
      <Input name="input" placeholder="Saisissez" type="text" />
    </TestFormComponent>
  );
});
