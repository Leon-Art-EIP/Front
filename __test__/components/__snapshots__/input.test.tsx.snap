// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input snapshot 1`] = `
<div
  className="flex flex-col gap-2 sm:min-w-min min-w-0 relative"
>
  <div
    className="relative"
  >
    <input
      autoComplete="off"
      className="sm:min-w-min min-w-0 px-6 py-4 w-full rounded placeholder:text-tertiary-hover pr-10"
      id="input"
      name="input"
      onChange={[Function]}
      placeholder="Saisissez"
      type="text"
    />
  </div>
</div>
`;
